manifests/master.pp

```diff
@@ -79,8 +80,13 @@
   if $::osfamily != 'RedHat' {
     fail("Cannot configure an IPA master server on ${::operatingsystem} operating systems. Must be a RedHat-like operating system.")
   }
+  if $dns == true {
+    $pkglist=[$ipa::master::svrdnspkg,$ipa::master::svrpkg]
+  } else {
+    $pkglist=$ipa::master::svrpkg
+  }
 
-  realize Package[$ipa::master::svrpkg]
+  realize Package[$pkglist]
```

Comment:
Wouldn't this cause a duplicate resource error since $svrpkg is realized twice? Also, it's not fully qualified...